pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -113,14 +128,13 @@ func (r *ReconcileBuildRun) Reconcile(request reconcile.Request) (reconcile.Resu
 	err = r.client.Get(context.TODO(), types.NamespacedName{Name: buildRun.Spec.BuildRef.Name, Namespace: buildRun.Namespace}, build)
 	if err != nil {
 		updateErr := r.updateBuildRunErrorStatus(buildRun, err.Error())
-		return reconcile.Result{}, fmt.Errorf("errors: %v %v", err, updateErr)
+		return reconcile.Result{}, handleError("Failed to fetch the Build instance", err, updateErr)
```

Comment:
In `updateBuildRunErrorStatus` method the currentt `err` (error) is passed along, and therefore should this same method be enhanced to merge incoming errors? Let me share a snippet to illustrate my suggestion:

```go
err = r.updateBuildRunErrorStatus(buildRun, err.Error())
return reconcile.Result{}, err
```

This same scenario happens in different places in the reconciliation logic, like [for instance here](https://github.com/redhat-developer/build/blob/4266ccdc217de941a141cd0af09badc875cb6b78/pkg/controller/buildrun/buildrun_controller.go#L166).

So, if `updateBuildRunErrorStatus` takes care of merging those errors -- the same logic than your `handleError` method uses -- we can avoid introducing one extra step to return an possible error.

Reply:
@otaviof Thank you for the suggestion.
I agree with you. However, the `handleError` method focus to cover the scenario that the `err` is not nil but `updateErr` is nil, like the example I describe here: https://github.com/redhat-developer/build/pull/220#issuecomment-634500297